From d0b28ac0bbb498d75add3c207fefa7fb6484a5bd Mon Sep 17 00:00:00 2001
From: "Christopher A. Williamson" <contact@chrisaw.io>
Date: Tue, 17 Mar 2026 23:24:50 +0000
Subject: [PATCH] Configure build for GrapheneOS GSI: apps and packages
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Adapt the AOSP build configuration for GrapheneOS GSI:

- Replace generic webview module with TrichromeWebView (Vanadium)
- Use GrapheneOS TrichromeChrome as browser (replaces Browser2)
- Remove QuickSearchBox (not in GrapheneOS)
- Remove Camera and Messaging from GSI product — users install
  from GrapheneOS App Store to save system image space
- Remove packages that are not useful or not desired in GSI builds:
  Calendar (deprecated, CalendarProvider kept), MusicFX, Auditor
  (requires hardware attestation), PdfViewerGOS, InfoApp,
  DeviceDiagnostics, EasterEgg, HardeningTestApp,
  LiveWallpapersPicker (deprecated), PhotoTable, WallpaperPickerGOS,
  ContactsGOS (use AOSP Contacts), BluetoothViewerGOS, TalkBackGOS
  (use standard TalkBack included in handheld_product.mk),
  GrapheneOS Updater (only useful for official GrapheneOS devices)
---
 target/product/base_system.mk         | 3 ---
 target/product/generic/Android.bp     | 3 +--
 target/product/gsi/Android.bp         | 7 ++-----
 target/product/handheld_product.mk    | 5 -----
 target/product/handheld_system.mk     | 3 ---
 target/product/handheld_system_ext.mk | 1 -
 6 files changed, 3 insertions(+), 19 deletions(-)

diff --git a/target/product/base_system.mk b/target/product/base_system.mk
index 2f2ccc9..33ce2a6 100644
--- a/target/product/base_system.mk
+++ b/target/product/base_system.mk
@@ -230,8 +230,6 @@ PRODUCT_PACKAGES += \
     mke2fs \
     mkfs.erofs \
     monkey \
-    misctrl \
-    mtectrl \
     ndc \
     netd \
     NetworkStack \
@@ -573,7 +571,6 @@ PRODUCT_PACKAGES_DEBUG := \
     unwind_info \
     unwind_reg_info \
     unwind_symbols \
-    HardeningTestApp \
 
 # The set of packages whose code can be loaded by the system server.
 PRODUCT_SYSTEM_SERVER_APPS += \
diff --git a/target/product/generic/Android.bp b/target/product/generic/Android.bp
index 1d7dcbf..083dd43 100644
--- a/target/product/generic/Android.bp
+++ b/target/product/generic/Android.bp
@@ -419,7 +419,7 @@ android_filesystem_defaults {
         ///////////////////////////////////////////
         // media_product
         ///////////////////////////////////////////
-        "webview",
+        "TrichromeWebView",
 
         ///////////////////////////////////////////
         // base_product
@@ -817,7 +817,6 @@ system_image_defaults {
                 "MediaProviderLegacy", // base_system
                 "MmsService", // handheld_system
                 "MtpService", // handheld_system
-                "MusicFX", // handheld_system
                 "NetworkStack", // base_system
                 "ONS", // telephony_system
                 "PacProcessor", // handheld_system
diff --git a/target/product/gsi/Android.bp b/target/product/gsi/Android.bp
index 3fa6e76..82d5705 100644
--- a/target/product/gsi/Android.bp
+++ b/target/product/gsi/Android.bp
@@ -129,11 +129,9 @@ android_filesystem_defaults {
         ///////////////////////////////////////////
         // gsi_product
         ///////////////////////////////////////////
-        "Browser2",
-        "Camera2",
+        "TrichromeChrome",
         "Dialer",
         "LatinIME",
-        "messaging",
         "apns-full-conf.xml",
         "frameworks-base-overlays",
     ],
@@ -187,13 +185,12 @@ android_system_image {
 
         // handheld_product
         "AvatarPicker",
-        "Calendar",
         "Contacts",
         "DeskClock",
         "Gallery2",
+        "LiveWallpapersPicker",
         "Music",
         "preinstalled-packages-platform-handheld-product.xml",
-        "QuickSearchBox",
         "SettingsIntelligence",
         "frameworks-base-overlays",
 
diff --git a/target/product/handheld_product.mk b/target/product/handheld_product.mk
index 2a4602d..b276cf9 100644
--- a/target/product/handheld_product.mk
+++ b/target/product/handheld_product.mk
@@ -23,17 +23,12 @@ $(call inherit-product, $(SRC_TARGET_DIR)/product/media_product.mk)
 # /product packages
 PRODUCT_PACKAGES += \
     AppStore \
-    Auditor \
-    Calendar \
-    Camera \
     Contacts \
     DeskClock \
     ExactCalculator \
     Gallery2 \
-    InfoApp \
     LatinIME \
     Music \
-    PdfViewerGOS \
     preinstalled-packages-platform-handheld-product.xml \
     SettingsIntelligence \
     SpeechServices \
diff --git a/target/product/handheld_system.mk b/target/product/handheld_system.mk
index 5c802e0..d2bd0b7 100644
--- a/target/product/handheld_system.mk
+++ b/target/product/handheld_system.mk
@@ -47,10 +47,8 @@ PRODUCT_PACKAGES += \
     CertInstaller \
     CredentialManager \
     DeviceAsWebcam \
-    DeviceDiagnostics \
     DocumentsUI \
     DownloadProviderUi \
-    EasterEgg \
     ExternalStorageProvider \
     FusedLocation \
     GmsCompat \
@@ -61,7 +59,6 @@ PRODUCT_PACKAGES += \
     ManagedProvisioning \
     MmsService \
     MtpService \
-    MusicFX \
     NetworkLocation \
     PacProcessor \
     preinstalled-packages-platform-handheld-system.xml \
diff --git a/target/product/handheld_system_ext.mk b/target/product/handheld_system_ext.mk
index 529fa6f..cf9b894 100644
--- a/target/product/handheld_system_ext.mk
+++ b/target/product/handheld_system_ext.mk
@@ -27,7 +27,6 @@ PRODUCT_PACKAGES += \
     Launcher3QuickStep \
     Provision \
     Settings \
-    SetupWizard2 \
     StorageManager \
     SystemUI \
     WallpaperCropper \
-- 
2.54.0

